com_plugin - Moved Filters + ordering

pe7er opened the pull request:
This PR is a replacement for "com_plugin - Moved Filters from sidebar to Search Tools #7699".

This PR will move the filters from sidebar to Search Tools
**and** adds Ordering + Direction dropdown options to Search Tools area.
Please use testing procedure of #7699

![com_plugins_filter_odering](https://cloud.githubusercontent.com/assets/1217850/10711829/4279b2d4-7a88-11e5-83a2-01ebb6404dcc.png)

**Note:** the **click-on-column-ordering** does **not work** anymore. If someone spots the cause of that issue, please help me to correct it. Thanks!

Commit:
com_plugin - Moved Filters + ordering

zero-24 commented:
@pe7er travis is failing here?

Commit:
changed spaces to tabs

Commit:
added space

pe7er commented:
@zero-24 No, I was failing regarding code style and of course Travis was not happy with that ;-)

jduerscheid commented:
I have tested this item :white_check_mark: successfully on f785e1af6020666f48378ae29bac61528d51db0b<br /><br />I tested it again sucessfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8146">issues.joomla.org/joomla-cms/8146</a>.</sub>

designbengel commented:
I have tested this item :white_check_mark: successfully on f785e1af6020666f48378ae29bac61528d51db0b<br /><br />Tested successful, ordering works. BTW Question: Why do we have everywhere - Select status - which shows "all" and additionally "All" - wouldn´t it be easier to have a "All Status" for select label ?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8146">issues.joomla.org/joomla-cms/8146</a>.</sub>

zero-24 commented:
RTC.

@committer please have a look into the comment by @pe7er bevor committing.

>Note: the click-on-column-ordering does not work anymore. If someone spots the cause of that issue, please help me to correct it. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8146">issues.joomla.org/joomla-cms/8146</a>.</sub>

zero-24 commented:
And the comment by @designbengel as well ;)

roland-d commented on administrator/components/com_plugins/models/plugins.php:
```diff
@@ -119,7 +119,13 @@ protected function getStoreId($id = '')
 	{
 		$search = $this->getState('filter.search');
-		$ordering = $this->getState('list.ordering', 'ordering');
+		$ordering = $this->getState('list.tableOrdering', 'ordering');
```
This change breaks the column header ordering. Column header uses the **ordering** variable not the **tableOrdering**.

roland-d commented on administrator/components/com_plugins/models/plugins.php:
```diff
@@ -140,7 +146,7 @@ protected function _getList($query, $limitstart = 0, $limit = 0)
 			}
 
-			$direction = ($this->getState('list.direction') == 'desc') ? -1 : 1;
+			$direction = ($this->getState('list.tableDirection') == 'desc') ? -1 : 1;
```
This change breaks the column header ordering. Column header uses the **ordering** variable not the **tableOrdering**.

roland-d commented on administrator/components/com_plugins/models/plugins.php:
```diff
@@ -162,7 +168,7 @@ protected function _getList($query, $limitstart = 0, $limit = 0)
 			}
 
-			$query->order($this->_db->quoteName($ordering) . ' ' . $this->getState('list.direction'));
+			$query->order($this->_db->quoteName($ordering) . ' ' . $this->getState('list.tableDirection'));
```
This change breaks the column header ordering. Column header uses the **ordering** variable not the **tableOrdering**.

roland-d commented:
@pe7er I commented on where it breaks. You may want to check if both filters can set the same value, so you don't need to change the model but just the template file. My 2ct.

Commit:
changed tableOrdering to ordering

pe7er commented:
Thanks @roland-d. 
Now the column ordering is working again

But the **Ordering & Direction dropdown** are **not working** anymore.

I tried to add the following to the populateState method but that did not work either:
 ```php 
$ordering = $this->getUserStateFromRequest($this->context . '.list.ordering', 'list_ordering', '', 'cmd');
$this->setState('list.ordering', $ordering);
```

designbengel commented:
what have you meant with column ordering? because on my site i thought it worked - but maybe i meant something else :)

pe7er commented:
Sorry for the confusion. With Column Ordering, I mean that **clicking** on the **Column headers will order** the results: {Ordering}, Status, Plugin Name, Type, Element, Access, ID (like it's currently working without this PR).

The dropdown boxes on the top right "Sort Table By" & "select the ordering" are not working anymore.

**Technical background info**: 
* I removed the code for the 3 boxes (Ordering, Direction & List Limit) from the view (/administrator/components/com_plugins/views/plugins/tmpl/default.php). 
* I added the functionality to the Search Tools (/administrator/components/com_plugins//models/forms/filter_plugins.xml)
* However, I cannot get the status of the "Sort Table By" & "select the ordering" dropdown boxes to work correctly in the _getList method.

Commit:
Changed Joomla.orderTable JavaScript

Commit:
fixed dropdown ordering boxes

pe7er commented:
With help from @roland-d (thanks!) I **fixed** the **dropdown boxes** on the top right **"Sort Table By"** & **"select the ordering"**. Thanks Roland!

Commit:
ordering dropdown & column ordering synchronized

Commit:
code style correction

pe7er commented:
Now the dropdown ordering & column header ordering are synchronized. If you click on a column header, the ordering dropdown should take over the same ordering value as the column header, and the direction dropdown box too. 

Huge thanks to @roland-d for helping me out with the JavaScript code needed to accomplish this.

Commit:
code style correction

roland-d committed:
Cleaned up the JavaScript and filled the form with the correct data

pe7er committed:
Merge pull request #13 from roland-d/remove-js

Cleaned up the JavaScript and filled the form with the correct data. Thanks @roland-d !

Commit:
Merge branch 'staging' of https://github.com/joomla/joomla-cms into search-tools-filters-plugins2

Commit:
Merge branch 'search-tools-filters-plugins2' of https://github.com/pe7er/joomla-cms into search-tools-filters-plugins2

Commit:
fix unit tests

pe7er commented:
Thanks @roland-d for unit test correction!

wilsonge commented on libraries/legacy/model/list.php:
```diff
@@ -457,6 +457,10 @@ protected function loadFormData()
 		}
 
+		// Set the selected filter values for pages that use the JLayouts for filtering
+		$data->list['sortTable'] = $this->state->{'list.ordering'};
```
$this->state->get('list.ordering');

Commit:
fix for unit test

pe7er commented:
Thanks @wilsonge for the correction to fix unit test error!

nonumber commented:
> nonumber: Wasn't the whole idea to move away from the Search Tools and use the side bar instead?
https://github.com/joomla/joomla-cms/pull/7699#issuecomment-131766601
> compojoom: @nonumber - that's what I'm wondering as well. Do we have a "design guide" or something? It would be stupid to have the filters once in search tools, then on the right. We should decide what we want to use and then stick to it.
https://github.com/joomla/joomla-cms/pull/7699#issuecomment-139985650

Commit:
fixed codestyle issue

Kubik-Rubik commented:
Restarted Travis checks.

dgt41 commented:
I have tested this item :white_check_mark: successfully on 45f7cb1c9af9ea5a406b89862e4f591a70cd01e8<br /><br />I would change the class for the pagination to `class="pagination pagination-toolbar center"` to get it centered, now is is left aligned. weird...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8146">issues.joomla.org/joomla-cms/8146</a>.</sub>

zero-24 commented:
Travis checks fails here: https://travis-ci.org/joomla/joomla-cms/jobs/89131184

wilsonge commented on administrator/components/com_plugins/models/fields/plugintype.php:
```diff
@@ -0,0 +1,44 @@
+require_once __DIR__ . '/../../helpers/plugins.php';
+
+/**
+ * Plugin Type Field class for the Joomla Framework.
```
I know it's being petty but this isn't from the framework any more - I guess this is a reference back to ye olde J1.5 framework :P

wilsonge committed:
Remove code

wilsonge committed:
Add custom loadFormData method

nibra commented on administrator/components/com_plugins/views/plugins/tmpl/default.php:
```diff
@@ -33,8 +33,8 @@
-		table = document.getElementById("sortTable");
-		direction = document.getElementById("directionTable");
+		table = document.getElementById("list_sortTable");
+		direction = document.getElementById("list_directionTable");
```
Where are the elements, these IDs are referring to?

wilsonge replied:
The XML form fields here https://github.com/joomla/joomla-cms/pull/8146/files#diff-66e42cf8aede4eac536f7ea226ceea5eR39

pe7er committed:
Merge pull request #16 from wilsonge/patch-11

Remove code, thanks @wilsonge !

wilsonge committed:
Remove test alterations

pe7er committed:
Merge pull request #17 from wilsonge/patch-11

Remove test alterations

pe7er commented:
Thanks for the corrections @wilsonge!

pe7er commented:
Thanks for testing @dgt41 
I think that the class="pagination pagination-toolbar center" can be added with separate PR?

designbengel commented:
I have tested this item :white_check_mark: successfully on f29115dea60de5620c0d6a119132d1b44004940b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8146">issues.joomla.org/joomla-cms/8146</a>.</sub>

dgt41 commented:
@test ok here as well.
@pe7er probably we have to align all the backend pagination either to left or to the center and stick to that rule…

nonumber commented:
Looks like I'm being ignored again (like in the other PR)
So again:

> @nonumber : 
Wasn't the whole idea to move away from the Search Tools and use the side bar instead?
https://github.com/joomla/joomla-cms/pull/7699#issuecomment-131766601

> @compojoom : 
@nonumber - that's what I'm wondering as well. Do we have a "design guide" or something? It would be stupid to have the filters once in search tools, then on the right. We should decide what we want to use and then stick to it.
https://github.com/joomla/joomla-cms/pull/7699#issuecomment-139985650

pe7er commented:
As far as I know, Search Tools are more advanced than the filters in the left sidebar: 
 * you can clear all filters with just one one click on "Clear", 
 * better separation of HTML output & function: the filters are defined in /models/forms/filter_somename.xml and not within the HTML of the default.php layout file.

wilsonge commented:
Imho I think the aim in this release has been to move things into search tools as per your guys code sprint for J3.2 (which only ever got half implemented in the CMS at that point). I think pretty much all the core components now use search tools

nonumber commented:
ok
